src/CodeGenerator/Generator/ApiGenerator.php

```diff
@@ -667,6 +667,14 @@ private function resultClassAddProperties(string $baseNamespace, bool $root, ?ar
             if ('structure' === $memberShape['type']) {
                 $this->doGenerateResultClass($baseNamespace, $parameterType);
                 $parameterType = $baseNamespace . '\\' . $this->safeClassName($parameterType);
+            } elseif ('map' === $memberShape['type']) {
```

Comment:
The ApiGenerator contains several occurence of `if type == 'structure` or `if type == 'list'`. Shouldn't we consider `map` also?

Reply:
We maybe should. I write an issue.

Reply:
See #100